### ISSUE TITLE
Exception while open any js or css file in node project folder?

### ISSUE THREAD

**Issue by babarqb** [Microsoft/nodejstools#1185]
Every time i open or create a file inside folder of nodjs project it through exception  but if i create or open file from root of project no exception throw. activity.xml  error is mention below

```
<entry>
    <record>819</record>
    <time>2016/07/31 19:19:05.919</time>
    <type>Error</type>
    <source>Editor or Editor Extension</source>
    <description>System.InvalidOperationException: Operation is not valid due to the current state of the object.&#x000D;&#x000A;   at Microsoft.VisualStudioTools.Project.Automation.OAProjectItem.get_Collection()&#x000D;&#x000A;   at Microsoft.VisualStudio.Web.Editor.Package.Lint.WebLinter.LinterUtilities.IsFileSupported(String fileName)&#x000D;&#x000A;   at Microsoft.VisualStudio.Web.Editor.Package.Lint.ErrorTagger.LintErrorTaggerProvider.CreateTagger[T](ITextBuffer textBuffer)&#x000D;&#x000A;   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.GatherTaggers(ITextBuffer textBuffer)</description>
  </entry>
```

- NTVS Version: 1.2
- Visual Studio Version : 2015 update 3 latest patch 
- Node.js Version:  6.1.3

**Comment by mjbvz** [comment]
Thanks for reporting this problem.

I'm unable to repo the issue in my testing and the `Microsoft.VisualStudio.Web.Editor.Package.Lint` error message suggests this is not an NTVS issue.

* Are there any additional steps to produce this error? Can you share the full path for a file that triggers this problem?
* Does it happen consistently?
* Does it happen in other projects? More interestingly, does it happen in non-NTVS projects?

Thanks.

**Comment by robbie-mac** [comment]
@mjbvz Hey Matt, I'm having the same issue popping up, and can answer a few of your questions:
* There aren't any steps to causing the error other than opening a .js file in an NTVS project from the solution explorer. The same .js file does not error when opened from the File|Open|File menu option, only when it is opened from the solution explorer. The .js file can be empty, or have code in it.
* It only happens the first time I open a .js file during a session. So if I open further .js files nothing happens.
* It does not happen for any other project types, just NTVS projects, and only started when I installed NTVS 1.2
I was previously using 1.1, however I can t recall if I had installed 1.2RC. Those facts are irrelevant  as I have isolated it by fresh installing VS and NTVS 1.2. Problem still exists. I am using Windows 10 x64 1511.

I have not seen this error for .css files, so I can not comment on that, however I am going to attach my whole activitylog.xml (minus the personally identifiable stuff. There are a few errors happening.
[ActivityLog.zip](https://github.com/Microsoft/nodejstools/files/421166/ActivityLog.zip)

**Comment by robbie-mac** [comment]
@babarqb @mjbvz Ok, so after a bunch of trying to fix the exception, and poking in things WAY over my head... not really, just too lazy to get into the nuts and bolts of this, I remembered that this type of problem poked its head up before (#371 might be similar issue).

So checking into maybe the chance that the error messages were in fact pointing me in the right direction, I began to look at my extensions. Some of them were disabled (Not by me) so I thought maybe something was wrong. I cleared my MEF Cache a few times, and uninstalled the disabled extensions, and cleared my cache again, and voila. No more exception, and everything works fine now.

So @babarqb , try clearing your MEF Cache ( Mads Kristensen has an [extension](https://visualstudiogallery.msdn.microsoft.com/22b94661-70c7-4a93-9ca3-8b6dd45f47cd) to help with that... that sounded like an Apple commercial ) give it a whirl. Then check your extensions to see if any are disabled, if there are remove them, clear your cache again and see if that helps.

**Comment by mleanos** [comment]
I'm experiencing this exact exception. I've cleared my MEF Cache, and performed the DevEnv.exe /setup task. I didn't notice any disabled extensions. The problem still persists.

Another thing that I've notice is that my automatic brace completion is not working on Node.js projects. However, it's working for .js files in my .NET projects. Not sure if this is related.

Visual Studio 2015 Community Update 3
NTVS 1.2.40726.00
Node v4.3.1

**Comment by robbie-mac** [comment]
Ive noticed that its a reccuring thing that keeps popping up. I think the problem stems from extensions getting updated impropely. Ive had to uninstall some extension... and then uninstall them again because 2 versions are registered. Once this is done then  the error goes away. I don't know why this happens or why my solution works... just that it does, and its a pain, and its not NTVS fault.

**Comment by BalassaMarton** [comment]
I have the same problem, with the following setup:
Visual Studio 2015 Professional Update 3
NTVS 1.2.40726.00
Node 6.5.0
Clearing the cache, reinstalling NTVS, Node, etc. didn't make it go away.

**Comment by sonealtx** [comment]
I'm having the same problem:

Visual Studio Enterprise 2015 Update 3
NTVS:  1.2.40726.00
Node: 4.6.0

I've tried clearing the cache, reinstalling NTVS, Node.  Deleting the folder: %localappdata%\Microsoft\VisualStudio\14.0,  Clearing the MEF Cache.

I've also noticed that my auto brace completion is not working nor the intellisense.
Thanks

**Comment by jawa-the-hutt** [comment]
Having the same problem. 

fresh load of Windows 10 1607 on a brand new laptop.
VSE 2015 Update 3
NTVS 1.2.40726.00
Node 6.8

Have had the problem for a couple of months now with the previous OS load, but was hoping it would go away with the new laptop.

**Comment by HoosierMike** [comment]
Same problem here, not good, we're screwed ...

**Comment by mjbvz** [comment]
Sorry for the delay handling this. I've been coordinating with the Web Tools team to fix this on their side, but also have merged a PR #1386 that should workaround this on the NTVS side. Try picking up the next dev build to try this out.
